File: app/src/main/java/com/maufonseca/mauriciocv/infrastructure/SearchEngine.java
package com.maufonseca.mauriciocv.infrastructure;

import com.maufonseca.mauriciocv.model.AppSnippet;
import com.maufonseca.mauriciocv.model.Snippet;

import java.util.ArrayList;

public class SearchEngine {

  public static ArrayList<Snippet> search(String query) {
    ArrayList<Snippet> response = new ArrayList<>();

    for (Snippet snippet : Snippet.getSynthesis()) {
      if(snippet.matchesQuery(query)) {
        response.add(snippet);
      }
    }

    for (Snippet snippet : Snippet.getSchool()) {
      if(snippet.matchesQuery(query)) {
        response.add(snippet);
      }
    }

    for (Snippet snippet : Snippet.getComplement()) {
      if(snippet.matchesQuery(query)) {
        response.add(snippet);
      }
    }

    for (Snippet snippet : WorkHistoryRequester.getWorkHistory()) {
      if(snippet.matchesQuery(query)) {
        response.add(snippet);
      }
    }

    for (Snippet snippet : AppSnippet.getApps()) {
      if(snippet.matchesQuery(query)) {
        response.add(snippet);
      }
    }

    return response;
  }
}

File: app/src/main/kotlin/com/maufonseca/mauriciocv/infrastructure/SearchEngine.kt
package com.maufonseca.mauriciocv.infrastructure

import com.maufonseca.mauriciocv.model.Snippet
import java.util.ArrayList

object SearchEngine {

  fun search(query: String): ArrayList<Snippet> = ArrayList(SynthesisRequester.synthesis.filter { it.matchesQuery(query) } +
        SchoolRequester.school.filter { it.matchesQuery(query) } +
        ComplementRequester.complement.filter { it.matchesQuery(query) } +
        WorkRequester.workHistory.filter { it.matchesQuery(query) } +
        AppsRequester.apps.filter { it.matchesQuery(query) })
}
